lib/switchtec.c

```diff
@@ -1333,36 +1340,58 @@ static int log_c_to_file(struct switchtec_dev *dev, int sub_cmd_id, int fd)
 	return 0;
 }
 
+static int log_ram_flash_to_file(struct switchtec_dev *dev,
+				 int gen5_cmd, int gen4_cmd, int gen4_cmd_lgcy,
+				 int fd, FILE *log_def_file,
+				 struct switchtec_log_file_info *info)
+{
+	int ret;
+
+	if (switchtec_is_gen5(dev)) {
+		return log_a_to_file(dev, gen5_cmd, fd, log_def_file,
+				     info);
+	} else {
+		ret = log_a_to_file(dev, gen4_cmd, fd, log_def_file,
+				    info);
+		if (!ret)
+			return 0;
+		
+		return log_a_to_file(dev, gen4_cmd_lgcy, fd,
+				     log_def_file, info);
+	}
+}
+
 /**
  * @brief Dump the Switchtec log data to a file
  * @param[in]  dev          - Switchtec device handle
  * @param[in]  type         - Type of log data to dump
  * @param[in]  fd           - File descriptor to dump the data to
  * @param[in]  log_def_file - Log definition file
+ * @param[out] info         - Log file information
  * @return 0 on success, error code on failure
  */
 int switchtec_log_to_file(struct switchtec_dev *dev,
 			  enum switchtec_log_type type,
 			  int fd,
-			  FILE *log_def_file)
+			  FILE *log_def_file,
+			  struct switchtec_log_file_info *info)
```

Comment:
Certainly this can be wrapped better.

Reply:
What do you suggest?

Reply:
Well log_def_file can easily be on the same line as fd....

But I'll often also use two tab indents instead of aligning to the open parenthesis to avoid too many lines (which is valid Linux style):

```
int switchtec_log_to_file(struct switchtec_dev *dev,
		enum switchtec_log_type type, int fd, FILE *log_def_file,
		struct switchtec_log_file_info *info)
```